### PR TITLE
Single entry

### PR DESCRIPTION
**Pull request by janphilipp-winkler**

### CODE REVIEW

File: components/ActivitiesList/ActivitiesList.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
+import activities from "@/activities.json";
+import * as Styled from "./ActivitiesList.styled";
+import { useState } from "react";
+import { motion } from "framer-motion";
+import NavButton from "../NavButton/NavButton";
+import useSWR from "swr";
+
+export default function ActivitiesList() {
+  const [filterPhrase, setFilter] = useState();
```
Comment (MerleMak) [inline components/ActivitiesList/ActivitiesList.js:9]:
```suggestion
  const [filterPhrase, setFilterPhrase] = useState();
```

File: components/ActivitiesList/ActivitiesList.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+      return true;
+    } else if (filterPhrase === "tools") {
+      return activity.hasOwnProperty("tool");
+    } else if (Array.isArray(activity.forEmotion)) {
+      return activity.forEmotion.includes(filterPhrase);
+    }
+  });
+
+  function handleFilter(event) {
+    setFilter(event.target.value);
```
Comment (MerleMak) [inline components/ActivitiesList/ActivitiesList.js:33]:
```suggestion
    setFilterPhrase(event.target.value);
```

File: pages/activities.js [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+            <ActivitiesForm handleShowForm={handleShowForm} />
+          ) : (
+            <ActivitiesList showForm={showForm} />
+          )}
+        </Page>
+        {showForm ? null : (
+          <Navigation>
+            <NavButton linkToPage={"/"}>enter a mood</NavButton>
+          </Navigation>
+        )}
```
Comment (MerleMak) [inline pages/activities.js:51]: why don't you integrate this ternary into the ternary above? in both cases you are checking if showForm is true, so you might as well combine these two :)

File: pages/activities.js [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+import { motion } from "framer-motion";
+
+export default function Activities() {
+  const [showForm, setShowForm] = useState(false);
+  function handleShowForm() {
+    setShowForm(!showForm);
+  }
+
+  return (
+    <>
```
Comment (MerleMak) [inline pages/activities.js:19]:
```suggestion
```

File: pages/activities.js [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+            <ActivitiesList showForm={showForm} />
+          )}
+        </Page>
+        {showForm ? null : (
+          <Navigation>
+            <NavButton linkToPage={"/"}>enter a mood</NavButton>
+          </Navigation>
+        )}
+      </Container>
+    </>
```
Comment (MerleMak) [inline pages/activities.js:53]:
```suggestion
```
the container is already the parent :)

File: components/ActivitiesForm/ActivitiesForm.js [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+    // const submittedActivityForm = {
+    //   activity: event.target.elements.Activity.value,
+    //   emoji: event.target.elements.Emoji.value,
+    //   text: event.target.elements.Description.value,
+    //   forEmotion: selectedEmotions,
+    // };
+    // setActivities([...activities, submittedActivityForm]);
+    // event.target.reset();
+    // setSelectedEmotions([]);
+    // console.log(activities);
```
Comment (MerleMak) [inline components/ActivitiesForm/ActivitiesForm.js:28]:
```suggestion
```
delete comments please :)

File: components/ActivitiesForm/ActivitiesForm.js [hunk 7]
```diff
@@ -0,0 +1,139 @@
...
+import { mutate } from "swr";
+
+export default function ActivitiesForm({ handleShowForm }) {
+  const [selectedEmotions, setSelectedEmotions] = useState([]);
+  const [activities, setActivities] = useState(initialActivities);
+  function handleCheckboxChange(emotion, isChecked) {
+    if (isChecked) {
+      setSelectedEmotions([...selectedEmotions, emotion]);
+    } else {
+      setSelectedEmotions(selectedEmotions.filter((e) => e !== emotion));
```
Comment (MerleMak) [inline components/ActivitiesForm/ActivitiesForm.js:14]:
```suggestion
      setSelectedEmotions(selectedEmotions.filter((selectedEmotion) => selectedEmotion !== emotion));
```

Comment (MerleMak) [reply components/ActivitiesForm/ActivitiesForm.js:14]: usually e is short for event, so i recommend to just name it fully to make it easier to read

File: components/EntriesList/EntriesList.js [hunk 8]
```diff
@@ -26,37 +37,84 @@ export default function EntriesList() {
...
+                      >
+                        no
+                      </Styled.DeleteAnswer>
+                    </>
+                  ) : (
+                    <>
+                      {" "}
+                      {/* <Styled.RoundButton as="a" href={`${entry._id}`}>
+                        <FiEye />
+                      </Styled.RoundButton> */}
```
Comment (MerleMak) [inline components/EntriesList/EntriesList.js:101]:
```suggestion
```

File: components/EntriesList/EntriesList.js [hunk 9]
```diff
@@ -26,37 +37,84 @@ export default function EntriesList() {
...
+                      <Styled.RoundButton
+                        as="a"
```
Comment (MerleMak) [inline components/EntriesList/EntriesList.js:103]:
```suggestion
```

File: components/EntriesList/EntriesList.js [hunk 10]
```diff
@@ -26,37 +37,84 @@ export default function EntriesList() {
...
+              exit={{ opacity: 0, scale: 0.8, transition: { duration: 0.3 } }}
+            >
+              <Styled.Card onClick={() => router.push(`${entry._id}`)}>
+                <Styled.AnimationContainer>
+                  <div
+                    style={{
+                      backgroundColor: entry.color,
+                      height: "100%",
+                      width: "100%",
+                    }}
```
Comment (MerleMak) [inline components/EntriesList/EntriesList.js:55]: you hopeI don't notice? 🤣😏

File: pages/activities.js [hunk 11]
```diff
@@ -0,0 +1,85 @@
...
+      <Container>
+        <Page>
+          <Header>
+            <GradientText>
+              <Guide text={"activities"} />
+            </GradientText>
+            <NewActivity onClick={handleShowForm}>
+              <motion.div
+                animate={showForm ? { rotate: 45 } : {}}
+                style={{ height: "22px" }}
```
[the remark this answers is not in the document]
Comment (MerleMak) [reply pages/activities.js:29]: for the motion div i would make the only exception if its too much of a hassle. but only here!

**Review by MerleMak — COMMENTED** [review]
Really nice Jan Philipp! 
I left you some comments, but really really good job! 🚀

Keep it up!

File: activities.json [hunk 12]
```diff
@@ -0,0 +1,108 @@
...
+    "emoji": "😃",
+    "description": "hold your smile for 10 seconds and uplift your mood",
+    "forEmotion": ["sadness", "anger", "fear"],
+    "tool": true,
+    "link": "/smiletrainer"
+  },
+  {
+    "title": "Inspirational Quotes",
+    "emoji": "🌟📖",
+    "description": "hold your smile for 10 seconds and uplift your mood",
```
[the remark this answers is not in the document]
Comment (janphilipp-winkler) [reply activities.json:13]: chat gpt klärt ;)

File: components/EntriesList/EntriesList.js (hunk 10 above)
Comment (janphilipp-winkler) [reply components/EntriesList/EntriesList.js:55]: style={{inlineStyling: `hidden from ${Merle}`}} // NOT WORKING!!!

File: pages/activities.js (hunk 3 above)
Comment (janphilipp-winkler) [reply pages/activities.js:51]: es fiel ihm wie schuppen von den augen...

File: pages/activities.js (hunk 11 above)
[the remark this answers is not in the document]
Comment (janphilipp-winkler) [reply pages/activities.js:29]: this element will be removed in the next PR anyways ;)

**Pull request merged**